cpp/include/cudf/join.hpp

```diff
@@ -396,6 +395,39 @@ std::unique_ptr<cudf::table> left_semi_join(
   null_equality compare_nulls         = null_equality::EQUAL,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Returns a vector of bools indicating whether there exist matching row(s) on the right
+ *        table for each row on the left table.
+ *
+ * Given two tables, each row in the `left_keys` table is search for matching rows in the
+ * `right_keys` table. The search result (found or not) is written into the corresponding row of the
+ * output array.
+ *
+ * The resulting vector describe exactly which rows of the left table should be returned from the
+ * query `left_semi_join(left_keys, right_keys, compare_nulls, nan_equality::UNEQUAL)`.
+ *
+ * @code{.pseudo}
+ * left_keys: {{0, 1, 2}}
+ * right_keys: {{5, 4, 1, 2, 3}}
+ * output: {false, true, true}
+ * @endcode
+ *
+ * @param left_keys The left table
+ * @param right_keys The right table
+ * @param compare_nulls Control whether null keys should be considered as matching or not
+ * @param compare_nans Control whether NaN keys in floating-point data should be considered as
+ *        matching or not
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @return A vector of bools indicating if each row in `left_keys` has matching row(s) in
+ *         `right_keys`
+ */
+rmm::device_uvector<bool> left_semi_join_contains(
```

Comment:
> This function belongs in the search namespace (search.hpp).

I'm happy with this.

Regarding public API, I think if there is any external FEA request then we can make this happen easily. The existing public API is dealing with `column_view` input thus it needs to return a column with null mask. Here this function is dealing with table input, the result does not have any null mask thus a boolean vector should be enough.

I agree that in the meantime we just create a detail API. This will be used immediately in my other dependent PR.

Reply:
This is also the main point that I need to have early review. Thanks @vyasr.

Reply:
Probably out of scope for this specific PR, but once it's done we should probably benchmark `contains(column_view const&, column_view const&)` against the new `contains(table_view const&, table_view const&)`. If the latter is faster, we could unify the implementations since the latter is a strict generalization of the former.